src/Core/Services/SyncService.cs

```diff
@@ -382,5 +383,39 @@ private async Task SyncSendsAsync(string userId, List<SendResponse> response)
                 new Dictionary<string, SendData>();
             await _sendService.ReplaceAsync(sends);
         }
+
+        private async Task SyncPasswordlessLoginRequestsAsync(string userId)
+        {
+            // if the user has not enabled passwordless logins ignore requests
+            if (!await _stateService.GetApprovePasswordlessLoginsAsync(userId))
+            {
+                return;
+            }
+
+            var loginRequests = await _apiService.GetAuthRequestAsync();
+            if (loginRequests == null || !loginRequests.Any())
+            {
+                return;
+            }
+
+            var validLoginRequest = loginRequests.Where(l => l.RequestApproved == null
+                                                 && l.ResponseDate == null
+                                                 && l.CreationDate.ToUniversalTime().AddMinutes(Constants.PasswordlessNotificationTimeoutInMinutes) > DateTime.UtcNow)
+                                     .OrderByDescending(x => x.CreationDate)
+                                     .FirstOrDefault();
+
+            if (validLoginRequest is null)
+            {
+                return;
+            }
+
+            await _stateService.SetPasswordlessLoginNotificationAsync(new PasswordlessRequestNotification()
+            {
+                Id = validLoginRequest.Id,
+                UserId = userId
+            });
+
+            _messagingService.Send(Constants.PasswordlessLoginRequestKey);
+        }
```

Comment:
We need to revise the `FullSyncAsync` method regarding exception handling and logging.
But for now could you add try...catch to your method that logs the exception and doesn't rethrow so that if an error happens syncing the passwordless login requests it doesn't break the full sync and given that this one doesn't rely, afaik, on the date of the last sync we can do it